compiler/rustc_mir/src/transform/instcombine.rs

```diff
@@ -121,4 +134,36 @@ impl<'tcx, 'a> InstCombineContext<'tcx, 'a> {
             }
         }
     }
+
+    fn find_copynonoverlapping(&self, stmt: &Statement<'tcx>) -> Option<(Local, Local)> {
+        if let StatementKind::CopyNonOverlapping(box CopyNonOverlapping { src, dst, count }) =
+            &stmt.kind
+        {
+            let constant = count.constant()?;
+            if let Some(1) = constant.literal.try_eval_usize(self.tcx, self.param_env) {
+                let src = src.place()?.as_local()?;
+                let dst = dst.place()?.as_local()?;
+
+                return Some((src, dst));
+            }
+        }
+
+        None
+    }
+
+    /// Transform "CopyNonOverlapping(src, dst, 1)" ==> "*dst = *src".
+    fn combine_copynonoverlapping(&self, stmt: &mut Statement<'tcx>) {
+        if let Some((src, dst)) = self.find_copynonoverlapping(stmt) {
+            if !self.should_combine_stmt(&stmt.source_info, &stmt.kind) {
+                return;
+            }
+
+            let deref_projection = self.tcx._intern_place_elems(&[PlaceElem::Deref]);
```

Comment:
There is `TyCtxt::mk_place_deref`, which would also apply uniformly to all places (not just locals).